source/multigrid/mg_transfer_internal.cc

```diff
@@ -564,6 +564,20 @@ namespace internal
                     continue;
                   cell->child(c)->get_mg_dof_indices(local_dof_indices);
 
+                  // Replace any indices of identity constrained dofs with the
+                  // indices of the dofs in which they are constrained
+                  if (mg_constrained_dofs != nullptr)
+                    for (unsigned int i=0; i<local_dof_indices.size(); ++i)
+                      if (mg_constrained_dofs->get_level_constraint_matrix(level).
+                          is_identity_constrained(local_dof_indices[i]))
+                        {
+                          Assert(mg_constrained_dofs->get_level_constraint_matrix(level).
+                                 get_constraint_entries(local_dof_indices[i])->size() == 1,
+                                 ExcInternalError());
+                          local_dof_indices[i] = mg_constrained_dofs->get_level_constraint_matrix(level).
+                                                 get_constraint_entries(local_dof_indices[i])->front().first;
+                        }
+
```

Comment:
Yes, I ran with the modified step 37 and applying the constraints on the coarsest level drops the iteration count from 8 to 7. I used degree 5.